### PR TITLE
Added auth token support for saas

### PR DESCRIPTION
**Pull request by rakeshv247**
Added support for authentication token which is primarily used for entry level authentication of a customer (contract) in SaaS

### CODE REVIEW

**Comment by dohnto** [comment]
Is there any other token except bearer? If there is not, why not pre-fill the "Bearer: " same se `kubectl` does.

**Comment by rakeshv247** [comment]
Yeah that would be a good idea...! I can do that

File: speechmatics/cli.py [hunk 1]
```diff
@@ -568,6 +570,16 @@ def parse_args(args=None):
         help="File(s) to process"
     )
 
+    transcribe_subparser.add_argument(
+        "--auth-token",
+        type=str,
+        help=(
+            "Pre shared bearerauthentication token to authorize the client "
```
Comment (owaisaamir) [inline speechmatics/cli.py:577]:
```suggestion
            "Pre shared authentication token to authorize the client "
```

File: speechmatics/cli.py [hunk 2]
```diff
@@ -568,6 +570,16 @@ def parse_args(args=None):
         help="File(s) to process"
     )
 
+    transcribe_subparser.add_argument(
+        "--auth-token",
+        type=str,
+        help=(
+            "Pre-shared authentication token to authorize the client "
+            "to use the RT services."
+            "eg. Bearer xyz=="
```
Comment (dohnto) [inline speechmatics/cli.py:579]:
```suggestion
            "eg. xyz=="
```

File: speechmatics/cli.py [hunk 3]
```diff
@@ -568,6 +570,16 @@ def parse_args(args=None):
         help="File(s) to process"
     )
 
+    transcribe_subparser.add_argument(
+        "--auth-token",
+        type=str,
+        help=(
+            "Pre-shared authentication token to authorize the client "
+            "to use the RT services."
```
Comment (viren-nadkarni) [inline speechmatics/cli.py:578]: This applies to RT SaaS only. Please mention it here.

File: speechmatics/models.py [hunk 4]
```diff
@@ -95,6 +95,9 @@ class ConnectionSettings:
     ping_timeout_seconds: float = 60
     """Ping-pong timeout in seconds."""
 
+    auth_token: str = None
+    """auth token to authenticate customer"""
```
Comment (viren-nadkarni) [inline speechmatics/models.py:99]: Please elaborate. This only applies to RT SaaS.

**Comment by viren-nadkarni** [comment]
Remember to `make docs` and commit the changes

**Review by viren-nadkarni — APPROVED** [review]

**Pull request merged**